Practical Example/RWTakeHome/RWTakeHome/ViewControllers/TutorialViewController.swift

```diff
@@ -0,0 +1,123 @@
+//
+//  ViewController.swift
+//  RWTakeHome
+//
+//  Created by Nick Nguyen on 10/7/20.
+//
+
+import UIKit
+
+class TutorialViewController: UIViewController {
+
+  //MARK:- Properties
+
+  let networkService = NetworkService()
+  var tutorials: [Tutorial] = []
+  var videos: [Tutorial] = []
+  var articles: [Tutorial] = []
+  private let cache = Cache<String, Data>()
+  private let photoFetchQueue = OperationQueue()
+  private var operations = [String: Operation]()
+
+  @IBOutlet weak var collectionView: UICollectionView! {
+    didSet { collectionView.dataSource = self }
+  }
+
+  //MARK:- IBActions
+  @IBAction func segmentedControlValueChanged(_ sender: UISegmentedControl) {
+    
+    switch sender.selectedSegmentIndex {
+    case 0: // Article
+
+      self.tutorials = self.articles.sorted(by: <)
+
+      self.reloadDataOnMainThread()
+
+    case 1: // Video
+
+      self.tutorials = self.videos.sorted(by: <)
+
+      self.reloadDataOnMainThread()
+
+    default: // All
+
+      self.tutorials = self.videos.sorted(by: <) + self.articles.sorted(by: <)
+      
+      self.reloadDataOnMainThread()
```

Comment:
You can move the reloading to beneath the switch statement and remove it from each case to avoid repeating it in every case.